src/plugins/kibana_utils/common/url/encode_uri_query.ts

```diff
@@ -28,11 +31,27 @@
  *    sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
  *                     / "*" / "+" / "," / ";" / "="
  */
-export function encodeQueryComponent(val: string, pctEncodeSpaces = false) {
+export function encodeUriQuery(val: string, pctEncodeSpaces = false) {
   return encodeURIComponent(val)
     .replace(/%40/gi, '@')
     .replace(/%3A/gi, ':')
     .replace(/%24/g, '$')
     .replace(/%2C/gi, ',')
+    .replace(/%3B/gi, ';')
     .replace(/%20/g, pctEncodeSpaces ? '%20' : '+');
 }
+
+export const encodeQuery = (
+  query: ParsedQuery,
+  encodeFunction: (val: string, pctEncodeSpaces?: boolean) => string = encodeUriQuery
+) =>
+  transform(query, (result, value, key) => {
+    if (key) {
+      const singleValue = Array.isArray(value) ? value.join(',') : value;
```

Comment:
Isn't this the same as `arrayFormat: 'comma'`?

```js
queryString.stringify({ foo: [1, 2, 3] }, { arrayFormat: 'comma'} );
//=> 'foo=1,2,3'
```

Reply:
@sqren yes, if you don't need to encode your object using `encodeQuery` method you can definitely use `{ arrayFormat: 'comma'}`. Now you can do it cause all plugins use `query-string` library directly without any wrappers.  Thank you for your suggestion.

I added this line, because initially the `encodeUriQuery` method worked only with the `string` type, but should also work fine with `string arrays`.

If you have any suggestions how to modify that code to use `{ arrayFormat: 'comma'} `please let me know.